bootloaders/eboot/eboot.c

```diff
@@ -22,7 +22,7 @@ extern void ets_wdt_enable(void);
 extern void ets_wdt_disable(void);
 
 // Converts bit of a string into a uint32
-#define S(a,b,c,d) ( (((uint32_t)a) & 0xff) | (((uint32_t)b) << 8) | (((uint32_t)c) << 16) | (((uint32_t)d)<<24) )
+#define S(a,b,c,d) ( (((int32_t)a) & 0xff) | (((uint32_t)b) << 8) | (((uint32_t)c) << 16) | (((uint32_t)d)<<24) )
```

Comment:
Sorry, didn't notice this change before.  As @d-a-v said, can you please undo the single uint32_t->int conversion here?